cmd/zpool/zpool_main.c

```diff
@@ -6029,51 +6032,63 @@ print_scan_status(pool_scan_stat_t *ps)
 
 	assert(ps->pss_state == DSS_SCANNING);
 
-	/*
-	 * Scan is in progress.
-	 */
+	/* Scan is in progress. Resilvers can't be paused. */
 	if (ps->pss_func == POOL_SCAN_SCRUB) {
 		if (pause == 0) {
 			(void) printf(gettext("scrub in progress since %s"),
 			    ctime(&start));
 		} else {
-			char buf[32];
-			struct tm *p = localtime(&pause);
-			(void) strftime(buf, sizeof (buf), "%a %b %e %T %Y", p);
-			(void) printf(gettext("scrub paused since %s\n"), buf);
-			(void) printf(gettext("\tscrub started on   %s"),
+			(void) printf(gettext("scrub paused since %s"),
+			    ctime(&pause));
+			(void) printf(gettext("\tscrub started on %s"),
 			    ctime(&start));
 		}
 	} else if (ps->pss_func == POOL_SCAN_RESILVER) {
 		(void) printf(gettext("resilver in progress since %s"),
 		    ctime(&start));
 	}
 
-	examined = ps->pss_examined ? ps->pss_examined : 1;
+	scanned = ps->pss_examined;
+	pass_scanned = ps->pss_pass_exam;
+	issued = ps->pss_issued;
+	pass_issued = ps->pss_pass_issued;
 	total = ps->pss_to_examine;
-	fraction_done = (double)examined / total;
 
-	/* elapsed time for this pass */
+	/* we are only done with a block once we have issued the IO for it */
+	fraction_done = (double)issued / total;
+
+	/* elapsed time for this pass, rounding up to 1 if it's 0 */
 	elapsed = time(NULL) - ps->pss_pass_start;
 	elapsed -= ps->pss_pass_scrub_spent_paused;
-	elapsed = elapsed ? elapsed : 1;
-	pass_exam = ps->pss_pass_exam ? ps->pss_pass_exam : 1;
-	rate = pass_exam / elapsed;
-	rate = rate ? rate : 1;
-	mins_left = ((total - examined) / rate) / 60;
-	hours_left = mins_left / 60;
-
-	zfs_nicebytes(examined, examined_buf, sizeof (examined_buf));
+	elapsed = (elapsed != 0) ? elapsed : 1;
+
+	scan_rate = pass_scanned / elapsed;
+	issue_rate = pass_issued / elapsed;
+
+	if (issue_rate != 0) {
+		hours_left = ((total - issued) / issue_rate) / 60 / 60;
+		mins_left = (((total - issued) / issue_rate) / 60) % 60;
+	} else {
+		hours_left = UINT64_MAX;
+		mins_left = UINT64_MAX;
+	}
+
+	/* format all of the numbers we will be reporting */
+	zfs_nicebytes(scanned, scanned_buf, sizeof (scanned_buf));
+	zfs_nicebytes(issued, issued_buf, sizeof (issued_buf));
 	zfs_nicebytes(total, total_buf, sizeof (total_buf));
+	zfs_nicebytes(scan_rate, srate_buf, sizeof (srate_buf));
+	zfs_nicebytes(issue_rate, irate_buf, sizeof (irate_buf));
 
 	/*
 	 * do not print estimated time if hours_left is more than 30 days
 	 * or we have a paused scrub
 	 */
+
 	if (pause == 0) {
-		zfs_nicebytes(rate, rate_buf, sizeof (rate_buf));
-		(void) printf(gettext("\t%s scanned out of %s at %s/s"),
-		    examined_buf, total_buf, rate_buf);
+		(void) printf(gettext("\t%s scanned at %s/s, "
+		    "%s issued at %s/s, %s total"),
+		    scanned_buf, srate_buf, issued_buf, irate_buf, total_buf);
```

Comment:
This line will exceed 80 characters, it will need to be shortened.

`	5.10T scanned at 1.46G/s, 5.94M issued at 29.7K/s, 8.56T total, (scan is slow, no estimated time)`

Reply:
Can you suggest a format then? I would like to break as few scripts as possible, while including the new statistics. I could move the time to completion to the next line, but I'm not sure if thats a good idea.

Reply:
>  would like to break as few scripts as possible

Always a good idea, but unless we're willing to only add new fields to the end anyone scraping this information will need to update their scripts anyway.  So let's take this opportunity to restructure the output.

I suggest the following format.  It moves the estimated time down a line and breaks out days and seconds so it's more human readable.  

```
  pool: jet22-1
 state: ONLINE
  scan: scrub in progress since Mon Oct 30 09:53:19 2017
	9.08T scanned at 1.62G/s, 1.47T issued at 933M/s, 552T total
	0B repaired, 0.27% done, 7 days 3:56:10 to go
```

We should also should drop, or rephrase, the 'scan is slow, no estimated time' message since this PR redefines what is meant by 'scan'.